c10/core/DispatchKey.h

```diff
@@ -452,6 +447,27 @@ enum class DispatchKey : uint16_t {
   _SparsePrivateUse3,
   EndOfSparseBackends = _SparsePrivateUse3,
 
+  // ~~~~~~~~~~~~~~ "SparseCsr" Per-Backend Dispatch keys ~~~~~~~~~~~~~~~~~~~ //
+  // keys starting with an _ are not currently used,
```

Comment:
Why have these underscores? Why not just have them as proper keys even if no one uses them?
There is a risk that out of tree users might start registering kernels for `_SparseCsrXPU` and then we can never rename it properly?

Reply:
Good point on the BC risk.

I originally did it so we could have a reasonable idea of which keys were actually used vs. not used, but you're right - someone might just start using the `_SparseCsrXPU` key out-of-tree without us knowing anyway.

I'll remove all of the leading underscores from the keys as part of this PR too.